common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/AccessCondition.scala

```diff
@@ -23,4 +31,28 @@ object AccessStatus {
   case object Unavailable extends AccessStatus
 
   case object PermissionRequired extends AccessStatus
+
+  def apply(str: String): Result[AccessStatus] =
+    str.toLowerCase match {
+      case status if status.startsWith("open with advisory") =>
+        Right(AccessStatus.OpenWithAdvisory)
+      case status if status.startsWith("open") =>
+        Right(AccessStatus.Open)
+      case status if status.startsWith("restricted") =>
+        Right(AccessStatus.Restricted)
+      case status if status.startsWith("cannot be produced") =>
+        Right(AccessStatus.Restricted)
+      case status if status.startsWith("certain restrictions apply") =>
+        Right(AccessStatus.Restricted)
+      case status if status.startsWith("closed") =>
+        Right(AccessStatus.Closed)
+      case status if status.startsWith("missing") =>
+        Right(AccessStatus.Unavailable)
+      case status if status.startsWith("temporarily unavailable") =>
+        Right(AccessStatus.Unavailable)
+      case status if status.startsWith("permission required") =>
+        Right(AccessStatus.PermissionRequired)
+      case status =>
```

Comment:
Just wondering, do we want to throw in the model, or throw in the transformer? I can see why it works in both, but feels strange to do it in the model. Maybe returning an `Option[AccessStatus]` and then when empty throwing an error in the transformer?

Reply:
This functionality is also used by the Sierra transformer, hence why shared between both. Having a `Result` vs `Option` doesn't feel any different to me personally, and a `Result` is more descriptive about what went wrong.